src/main/java/io/xpring/xrpl/DefaultXpringClient.java

```diff
@@ -204,6 +207,41 @@ public RawTransactionStatus getRawTransactionStatus(String transactionHash) thro
         return new RawTransactionStatus(response);
     }
 
+    /**
+     * Retrieve the transaction history for an address.
+     *
+     * @param address: The address to retrieve transaction history for.
+     * @return: An array of {@link io.xpring.xrpl.Transaction}s for the account.
+     * @throws XpringException If the given inputs were invalid.
+     */
+    @Override
+    public io.xpring.xrpl.Transaction [] getTransactionHistory(String address) throws XpringException {
+        Objects.requireNonNull(address);
+
+        if (!Utils.isValidXAddress(address)) {
+            throw XpringException.xAddressRequiredException;
+        }
+
+        ClassicAddress classicAddress = Utils.decodeXAddress(address);
+
+        AccountAddress account = AccountAddress.newBuilder().setAddress(address).build();
+        GetAccountTransactionHistoryRequest request = GetAccountTransactionHistoryRequest.newBuilder()
+                .setAccount(account)
+                .build();
+
+        GetAccountTransactionHistoryResponse response = this.stub.getAccountTransactionHistory(request);
+
+        List<GetTransactionResponse> getTransactionResponseList = response.getTransactionsList();
+
+        List<ImmutableTransaction> transactionsList = getTransactionResponseList.stream()
+                .map(DefaultXpringClient::getTransactionResponseToImmutableTransaction)
+                .collect(Collectors.toList());
+
+        ImmutableTransaction [] transactions = new ImmutableTransaction[transactionsList.size()];
+        transactions = transactionsList.toArray(transactions);
```

Comment:
Could be wrong here, but I think you can skip the `List<ImmutableTransaction>` and go straight from a `Stream` to a `ImmutableTransaction[]`.

Looks something like:
```java
ImmutableTransaction[] transactions = getTransactionResponseList.stream()
                .map(DefaultXpringClient::getTransactionResponseToImmutableTransaction)
                .toArray(size -> new ImmutableTransaction[size])
```